x/wasm/internal/keeper/proposal_handler.go

```diff
@@ -0,0 +1,141 @@
+		}
+		switch c := content.(type) {
+		case *types.StoreCodeProposal:
+			return handleStoreCodeProposal(ctx, k, *c)
+		case *types.InstantiateContractProposal:
+			return handleInstantiateProposal(ctx, k, *c)
+		case *types.MigrateContractProposal:
+			return handleMigrateProposal(ctx, k, *c)
+		case *types.UpdateAdminProposal:
+			return handleUpdateAdminProposal(ctx, k, *c)
+		case *types.ClearAdminProposal:
+			return handleClearAdminProposal(ctx, k, *c)
+		default:
+			return sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "unrecognized wasm proposal content type: %T", c)
+		}
+	}
+}
+
+func handleStoreCodeProposal(ctx sdk.Context, k Keeper, p types.StoreCodeProposal) error {
+	if err := p.ValidateBasic(); err != nil {
+		return err
+	}
+
+	codeID, err := k.Create(ctx, p.Creator, p.WASMByteCode, p.Source, p.Builder)
+	if err != nil {
+		return err
+	}
+
+	ourEvent := sdk.NewEvent(
+		sdk.EventTypeMessage,
+		sdk.NewAttribute(sdk.AttributeKeyModule, types.ModuleName),
+		//sdk.NewAttribute(AttributeSigner, p.Creator.String()), // todo: creator is not signer. rename attribute?
+		sdk.NewAttribute(AttributeKeyCodeID, fmt.Sprintf("%d", codeID)),
+	)
+	ctx.EventManager().EmitEvent(ourEvent)
+	return nil
+}
+
+func handleInstantiateProposal(ctx sdk.Context, k Keeper, p types.InstantiateContractProposal) error {
+	if err := p.ValidateBasic(); err != nil {
+		return err
+	}
+
+	contractAddr, err := k.Instantiate(ctx, p.Code, p.Creator, p.Admin, p.InitMsg, p.Label, p.InitFunds)
+	if err != nil {
+		return err
+	}
+
+	ourEvent := sdk.NewEvent(
+		sdk.EventTypeMessage,
+		sdk.NewAttribute(sdk.AttributeKeyModule, types.ModuleName),
+		//sdk.NewAttribute(AttributeSigner, p.Creator.String()),
+		sdk.NewAttribute(AttributeKeyCodeID, fmt.Sprintf("%d", p.Code)),
+		sdk.NewAttribute(AttributeKeyContract, contractAddr.String()),
+	)
+	ctx.EventManager().EmitEvent(ourEvent)
+	return nil
+}
+
+func handleMigrateProposal(ctx sdk.Context, k Keeper, p types.MigrateContractProposal) error {
+	if err := p.ValidateBasic(); err != nil {
+		return err
+	}
+
+	res, err := k.migrate(ctx, p.Contract, p.Sender, p.Code, p.MigrateMsg, GovAuthorizationPolicy{})
+	if err != nil {
+		return err
+	}
+
+	ourEvent := sdk.NewEvent(
+		sdk.EventTypeMessage,
+		sdk.NewAttribute(sdk.AttributeKeyModule, types.ModuleName),
+		//sdk.NewAttribute(AttributeSigner, p.Creator.String()),
+		sdk.NewAttribute(AttributeKeyContract, p.Contract.String()),
+	)
+	ctx.EventManager().EmitEvents(append(res.Events, ourEvent))
+	return nil
+}
+
+func handleUpdateAdminProposal(ctx sdk.Context, k Keeper, p types.UpdateAdminProposal) error {
+	if err := p.ValidateBasic(); err != nil {
+		return err
+	}
+
+	if err := k.setContractAdmin(ctx, p.Contract, p.Sender, p.NewAdmin, GovAuthorizationPolicy{}); err != nil {
+		return err
+	}
+
+	ourEvent := sdk.NewEvent(
+		sdk.EventTypeMessage,
+		sdk.NewAttribute(sdk.AttributeKeyModule, types.ModuleName),
+		//sdk.NewAttribute(AttributeSigner, p.Creator.String()),
```

Comment:
There is nothing here for other governance contracts either?

I'm fine, as long as it looks like the other code they ship.

Reply:
🤔 I have not found anything in the other handlers:
* [community pool](https://github.com/cosmos/cosmos-sdk/blob/v0.38.3/x/distribution/keeper/proposal_handler.go)
* [Params](https://github.com/cosmos/cosmos-sdk/blob/v0.38.3/x/params/proposal_handler.go)
* [Upgrade](https://github.com/cosmos/cosmos-sdk/blob/v0.38.3/x/upgrade/handler.go)

What would make sense for our use cases? If somebody wants to watch all contract modifications for example then we should create an event.... it was just the attribute name "signer" that does not fit the the proposal context

Reply:
Fair enough, I agree with your decisions.